OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs

```diff
@@ -568,6 +568,97 @@ public int ControlNumbersToBeRequested(string productCode)
+                            OUTPUT inserted.PremiumID, inserted.PolicyID, inserted.PayDate INTO @tblPremiums(PremiumId, PolicyId, PayDate)
+                            SELECT PL.PolicyID, NULL PayerId, P.ReceivedAmount Amount, P.ReceiptNo Receipt, P.PaymentDate PayDate, N'M' PayType, GETDATE() ValidityFrom, -1 AuditUserId, 0 IsOffline, 0 IsPhotoFee
+                            FROM tblControlNumber CN
+                            INNER JOIN tblPayment P ON CN.PaymentID = P.PaymentID
+                            INNER JOIN tblPaymentDetails PD ON P.PaymentID  = PD.PaymentID
+                            INNER JOIN tblInsuree I ON I.CHFID = PD.InsuranceNumber
+                            INNER JOIN tblOfficer O ON P.OfficerCode = O.Code
+                            INNER JOIN tblPolicy PL ON PL.FamilyID = I.FamilyID 
+						                            AND PD.PolicyStage = PL.PolicyStage
+                            INNER JOIN tblProduct Prod ON PL.ProdID = Prod.ProdID 
+						                            AND PD.ProductCode = Prod.ProductCode
+                            WHERE P.PaymentID = @PaymentId
+                            AND PL.PolicyStatus = 1
+                            AND PD.PremiumID IS NULL
+                            AND CN.ValidityTo IS NULL
+                            AND P.ValidityTo IS NULL
+                            AND PD.ValidityTo IS NULL
+                            AND I.ValidityTo IS NULL
+                            AND PL.ValidityTo IS NULL
+                            AND Prod.ValidityTo IS NULL
+                            AND O.ValidityTo IS NULL;
+
+                            DECLARE @TotalPremium DECIMAL(18, 2),
+		                            @PolicyValue DECIMAL(18, 2),
+                                    @PremiumId INT = 0;
+
+                            SELECT TOP 1 @PremiumId PremiumID FROM @tblPremiums;
+
+                            SELECT @TotalPremium = SUM(PR.Amount) OVER(ORDER BY Pr.Amount), @PolicyValue = PL.PolicyValue
+                            FROM tblPremium PR
+                            INNER JOIN @tblPremiums dt ON PR.PolicyID = dt.PolicyID
+                            INNER JOIN tblPolicy PL ON PR.PolicyID = PL.PolicyID
+                            WHERE PR.ValidityTo IS NULL
+                            AND PL.ValidityTo IS NULL
+
+                            IF @TotalPremium >= @PolicyValue
+                            BEGIN
+	                            UPDATE PL SET PolicyStatus = 2, EffectiveDate = dt.PayDate
+	                            FROM tblPolicy PL 
+	                            INNER JOIN @tblPremiums dt ON PL.PolicyID = dt.PolicyId
+	                            WHERE PL.ValidityTo IS NULL;
+
+	                            INSERT INTO tblInsureePolicy(InsureeId, PolicyId, EnrollmentDate, StartDate, EffectiveDate,ExpiryDate, ValidityFrom, AuditUserId, isOffline)
```

Comment:
Isn't this already added when synchronizing the family?

Reply:
it's added only if the policy is active and premium exists. In our case there will be no premium and policy will be idle